validator/src/validation/length.rs

```diff
@@ -32,11 +34,33 @@ pub fn validate_length<T: HasLen>(
     true
 }
 
+pub trait ValidateLenght: HasLen {
```

Comment:
It seems like if the user wants to implement `ValidateLength` on their own type, they'd need to also implement `HaLen`?

Might it be simpler if `HasLen` is not involved?

Reply:
Typo on the name and it shouldn't use `HasLen` at all. Then we can just create a macro to generate all the impls we want for each type. With this trait, HasLen doesn't exist anymore